held_suarez/held_suarez.F90

```diff
@@ -0,0 +1,166 @@
+  real(kind_phys), parameter :: sigmab  =  0.7_kind_phys ! threshold sigma level
+  real(kind_phys), parameter :: t00     = 200._kind_phys ! minimum reference temperature
+  real(kind_phys), parameter :: kf      = 1._kind_phys/(86400._kind_phys*efoldf) ! 1./efolding_time for wind dissipation
+
+  real(kind_phys), parameter :: onemsig = 1._kind_phys - sigmab ! 1. - sigma_reference
+
+  real(kind_phys), parameter :: ka      = 1._kind_phys/(86400._kind_phys * efolda) ! 1./efolding_time for temperature diss.
+  real(kind_phys), parameter :: ks      = 1._kind_phys/(86400._kind_phys * efolds)
+
+  !!
+  !! Model constants, reset in init call
+  !!
+  real(kind_phys)              :: cappa = 2.0_kind_phys / 7.0_kind_phys  ! R/Cp
+  real(kind_phys)              :: cpair = 1004.0_kind_phys        ! specific heat of dry air (J/K/kg)
+  real(kind_phys)              :: psurf_ref = 0.0_kind_phys       ! Surface pressure
+  ! pref_mid_norm are layer midpoints normalized by surface pressure ('eta' coordinate)
+  real(kind_phys), allocatable :: pref_mid_norm(:)
+  integer               :: pver                     ! Num vertical levels
+
+
+
+!======================================================================= 
+contains
+!======================================================================= 
+
+  !> \section arg_table_held_suarez_1994_init  Argument Table
+  !! \htmlinclude held_suarez_1994_init.html
+  subroutine held_suarez_1994_init(cappa_in, cpair_in, psurf_ref_in, pref_mid_norm_in)
+    !! Dummy arguments
+    real(kind_phys), intent(in) :: cappa_in
+    real(kind_phys), intent(in) :: cpair_in
+    real(kind_phys), intent(in) :: psurf_ref_in
+    real(kind_phys), intent(in) :: pref_mid_norm_in(:)
+
+    pver = size(pref_mid_norm_in)
+    allocate(pref_mid_norm(pver))
+    cappa         = cappa_in
+    cpair         = cpair_in
+    psurf_ref     = psurf_ref_in
+    pref_mid_norm = pref_mid_norm_in
+
+  end subroutine held_suarez_1994_init
+
+  !> \section arg_table_held_suarez_1994_run  Argument Table
+  !! \htmlinclude held_suarez_1994_run.html
+  subroutine held_suarez_1994_run(ncol, clat, pmid, &
+       u, v, t, du, dv, s)
+
+    !
+    ! Input arguments
+    !
+    integer,  intent(in)  :: ncol             ! Num active columns
+    real(kind_phys), intent(in)  :: clat(:)      ! latitudes(radians) for columns
+    real(kind_phys), intent(in)  :: pmid(:,:) ! mid-point pressure
+    real(kind_phys), intent(in)  :: u(:,:)    ! Zonal wind (m/s)
+    real(kind_phys), intent(in)  :: v(:,:)    ! Meridional wind (m/s)
+    real(kind_phys), intent(in)  :: t(:,:)    ! Temperature (K)
+                                              !
+                                              ! Output arguments
+                                              !
+    real(kind_phys), intent(out) :: du(:,:)   ! Zonal wind tend
+    real(kind_phys), intent(out) :: dv(:,:)   ! Meridional wind tend
+    real(kind_phys), intent(out) :: s(:,:)    ! Heating rate
```

Comment:
Could we possibly rename the variable `s` to `ds`, to indicate that it is also a tendency (usually `s` by itself represents the dry static energy, not the heating rate).